src/include/sof/audio/codec_adapter/interfaces.h

```diff
@@ -19,8 +19,13 @@
 #include <sof/audio/codec_adapter/codec/dummy.h>
 #endif
 
+#if CONFIG_WAVES_CODEC
+#include <sof/audio/codec_adapter/codec/waves.h>
+#endif
+
 #define CADENCE_ID 0xCADE01
 #define DUMMY_ID   0xD03311
+#define WAVES_ID   0x574101
```

Comment:
why do we need this? 

We already have UUIDs in topology, why do we need yet another ID and yet another list of effects?

This looks like we replaced an enum with a 24-bit ID, but it still doesn't make this scalable.

Reply:
This is for the codec adapter which has a static uuid. @mrajwa can answer more. I am in favour of the wrapper as it helps optimize and provide some universal benefits to codec. In other words, this is the design of the codec_adapter.

Reply:
UUID is for the codec_adapter - the SOF component, the CODEC_ID is to distinguish different codecs that are used by codec_adapter.

Reply:
In what cases would the SOF component and codec ID be different? The cadence case showed above makes me this that there's a unique ID although that Cadence library contains multiple codecs. If I followed your logic, you would have one ID per codec...

The point of the UUID for components was to move away from the requirement to have a central list of IDs. You are bringing this back here, possibly at a different level, and I think it's a terrible idea. We've learned that once you reach ~10ish items the list starts to grow into an unmanageable state, with conflicts between different providers.

It's not just me thinking this way, I will also point to the Intel closed-source firmware where there is no requirement for the base firmware to track the IDs of third party IP. We have to find a different mechanism for the 3rd party component to request a unique ID from the core, or rely on UUIDs which will by construction be unique.

I would go one step further. We shouldn't prevent a 3rd party IP from developing code without ever releasing any code snippet or even request an ID from the SOF team. Having to disclose information by requesting a unique ID is already one bridge too far. The code is private, the SOF team does not have a need to know and shall not list any IDs. Not our business. We chose a permissive license on purpose, no one shall be required to disclose information on their SOF integration.

Reply:
I agree.


> In what cases would the SOF component and codec ID be different?

In that codec by itself is not an SoF component - meaning it needs an adapter and this is SoF component with UUID.

I agree that maintaining a long list of codecs ID is problematic. I think we can create macro inside codec_adapter which will do what `DECLARE_SOF_RT_UUID` does. Hence any codec which whish to use codec_adapter would declare its ID as well as interfaces. Does it sound ok for you?

Reply:
@mrajwa I don't want to prescribe a solution, I am happy with any mechanism that remove this central list and registers the codec_adapter with the base firmware. If we all agree on the direction I don't mind removing my objection and having this done in a follow-up PR.

Reply:
Yes sure thing, I already have one more change on my todo list for codec_adapter so will cover both soon. In the meantime I think we can accept the way this PR handles that.

Reply:
Solution discussed in TSC, will follow up there